pyuntl/untldoc.py

```diff
@@ -675,3 +676,42 @@ def find_untl_errors(untl_dict, **kwargs):
         'error_dict': error_dict,
     }
     return found_data
+
+
+def untl_dict_to_tuple(untl_dict):
+    """Convert values of untl dict to tuple form which will be later used for hashing."""
+    untl_tuple = {}
+    for elem in untl_dict:
+        for i, v in enumerate(untl_dict[elem]):
+            if isinstance(untl_dict[elem][i]['content'], dict):
+                untl_dict[elem][i]['content'] = [v for v in untl_dict[elem][i]['content'].items()]
+                untl_dict[elem][i]['content'].sort()
+            untl_dict[elem][i] = [v for v in untl_dict[elem][i].items()]
+        untl_tuple[elem] = untl_dict[elem]
+    return untl_tuple
+
+
+def generate_hash(input):
+    """Return an md5 hash of the output of untl_to_tuple_form."""
+    return hashlib.md5(repr(input).encode()).hexdigest()
+
+
+def untl_to_hash_dict(untl_elements, meaningfulMeta=True):
+    """Convert untl element to hashed untl dictionary."""
+    if meaningfulMeta:
+        remove_elements = []
+        for element in untl_elements.children:
+            if (
+                element.tag == 'meta'
+                and (element.qualifier == 'metadataModificationDate'
+                     or element.qualifier == 'metadataModifier')
+            ):
+                remove_elements.append(element)
+        for elem in remove_elements:
+            untl_elements.remove_child(elem)
```

Comment:
Hmm..what you said makes sense. Initially I thought we may want to use a method for removing children, the way we were adding them. But certainly removing them from untl_dict is safer. Do you think I can let the `remove_child` method be there, if it can be used somewhere else?

Reply:
I think at this point it makes sense to remove it because:
1. We don't currently have a use case for removing children
2. The content of the method would need to change because I don't know if checking `self.contained_children` makes sense--maybe it does? But even if the child is listed in `self.contained_children`, it doesn't mean the element has that child, and if it doesn't have the child, trying to remove it there will cause a `ValueError`
3. We would probably want to add a test, which I don't think it makes sense to go to the trouble of if we aren't going to be using the method